### PR TITLE
jbpm-tests: Add tests for Custom Assignment Strategy

### PR DESCRIPTION
**Pull request by jakubschwan**
One test fail due [JBPM-6042](https://issues.jboss.org/browse/JBPM-6042)
AssignmentStrategy interface shouldn't be in internal repo - for this is reported issue [JBPM-6039](https://issues.jboss.org/browse/JBPM-6039)

### CODE REVIEW

File: jbpm-human-task/jbpm-human-task-core/src/test/java/org/jbpm/services/task/assignment/CustomAssignmentStrategyTest.java [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+            emf.close();
+        }
+        if (pds != null) {
+            pds.close();
+        }
+    }
+
+    @Test
+    public void testAssignTaskWithMultiActor() {
+        final String taskString = createTaskString(MULTI_ACTOR_ASSIGNMENTS, "MultiActorCustomTask");//TODO update name
```
Comment (rsynek) [inline jbpm-human-task/jbpm-human-task-core/src/test/java/org/jbpm/services/task/assignment/CustomAssignmentStrategyTest.java]: TODO?

File: jbpm-human-task/jbpm-human-task-core/src/test/java/org/jbpm/services/task/assignment/strategy/CustomStrategy.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CustomStrategy implements AssignmentStrategy {
+
+    private static final Logger logger = LoggerFactory.getLogger(CustomStrategy.class);
+    private static final String IDENTIFIER = "Custom";
+    private static final String ASSIGN_TO_USER = "Darth Vader";
+
+    public CustomStrategy() {
```
Comment (rsynek) [inline jbpm-human-task/jbpm-human-task-core/src/test/java/org/jbpm/services/task/assignment/strategy/CustomStrategy.java]: The constructor is not necessary here.

Comment (rsynek) [inline jbpm-human-task/jbpm-human-task-core/src/test/java/org/jbpm/services/task/assignment/strategy/CustomStrategy.java]: But you can e.g. use a constructor taking "assign to user" as a parameter to make the strategy usable for tests with different uses as well.

**Comment by rsynek** [comment]
+1 to merge after resolving discovered issues.

**Comment by tsurdilo** [comment]
@jakubschwan I left comment on JBPM-6042 -- could you please help me understand because I think that test method testAssignTaskNotForUser is correctly failing. Thanks.

**Comment by tsurdilo** [comment]
@jakubschwan @rsynek I have added fix for JBPM-6042 and have included your tests from this PR in it (I hope that is ok) since they test the fix specifically. PR here : https://github.com/kiegroup/jbpm/pull/868

Please review if you can. thanks.

**Comment by mswiderski** [comment]
closing as #868 included these tests

**Pull request closed without merging**